proposed/2021/MachineReadableOutput.md

```diff
@@ -0,0 +1,821 @@
+...
+
+```
+
+#### `> dotnet list package --include-transitive --format json`
+
+```json
+{
+  "version": 1,
+  "projects": {
+    "MyProjectA": [
+      {
+        "framework": "netcoreapp3.1",
+        "topLevelPackages": [
+          {
+            "id": "Microsoft.Extensions.Primitives",
+            "requestedVersion": "[1.0.0, 5.0.0]",
+            "resolvedVersion": "1.0.0"
+
+          },
+          {
+            "id": "NuGet.Commands",
+            "requestedVersion": "4.8.0-preview3.5278",
+            "resolvedVersion": "4.8.0-preview3.5278"
+          },
+          {
+            "id": "Text2Xml.Lib",
+            "requestedVersion": "[1.1.2, 2.0.0)",
+            "resolvedVersion": "1.1.2"
+          }
+        ],
+        "transitivePackages": [
+          {
+            "id": "Microsoft.CSharp",
+            "resolvedVersion": "4.0.1"
+          },
+          {
+            "id": "Microsoft.NETCore.Platforms",
+            "resolvedVersion": "1.1.0"
+          },
+  ...
+        ]
+      }
+    ],
+    "MyProjectB": [
+      {
+        "framework": "netcoreapp3.1",
+        "topLevelPackages": [
+          {
+            "id": "NuGet.Commands",
+            "requestedVersion": "4.8.0-preview3.5278",
+            "resolvedVersion": "4.8.0-preview3.5278"
+          },
+          {
+            "id": "Text2Xml.Lib",
+            "requestedVersion": "1.1.2",
+            "resolvedVersion": "1.1.2"
+          }
+        ],
+        "transitivePackages": [
+          {
+            "id": "Microsoft.CSharp",
+            "resolvedVersion": "4.0.1"
+          },
+          {
+            "id": "Microsoft.NETCore.Platforms",
+            "resolvedVersion": "1.1.0"
+          },
+  ...
+        ]
+      },
+      {
+        "framework": "net5.0",
+        "topLevelPackages": [ 
+          {
+            "id": "NuGet.Commands",
+            "requestedVersion": "4.8.0-preview3.5278",
+            "resolvedVersion": "4.8.0-preview3.5278"
+          },
+          {
+            "id": "Text2Xml.Lib",
+            "requestedVersion": "1.1.2",
+            "resolvedVersion": "1.1.2"
+          }
+        ],
+        "transitivePackages": [
+          {
+            "id": "Microsoft.CSharp",
+            "resolvedVersion": "4.0.1"
+          },
+          {
+            "id": "Microsoft.NETCore.Platforms",
+            "resolvedVersion": "1.1.0"
+          },
+  ...
+        ]
+      }
+    ]
+  }
+}
+```
+
+#### `--all` option
+
+Ability to use new `--all` option in conjunction with `--format` option shows all deprecated, vulnerable, and outdated top level and transitive packages. Users can form their own json as a last resort, instead of your team having to respond to every possible user need.  It would result in if `[--deprecated],[--include-transitive],[--outdated],[--vulnerable]` are combined, currently options '--outdated', '--deprecated' and '--vulnerable' cannot be combined on cli.
+
```

Comment:
![image](https://user-images.githubusercontent.com/8766776/145881197-d6ee5b58-55db-4828-a384-2c39b2b34142.png)
IDK, why it's designed that way for cli.